lib/sqlitex/row.ex

```diff
@@ -22,10 +22,21 @@ defmodule Sqlitex.Row do
     nil
   end
 
+  # date is empty ""
+  defp translate_value({date, "date"}) when date=="", do: to_date(date)
+
   defp translate_value({date, "date"}) when is_binary(date), do: to_date(date)
 
+  # time is empty ""
+  defp translate_value({time, "time"}) when time=="", do: to_time(time)
```

Comment:
You're still calling `to_time` with the empty string here, so this isn't much different from the original code.  You should maybe just return nil?